app/editor/src/features/navbar/NavBar.tsx

```diff
@@ -7,8 +7,47 @@ import { Row } from 'tno-core/dist/components/flex/row';
  */
 export const NavBar: React.FC = () => {
   const [activeHover, setActiveHover] = useState<string>('');
+
+  let isHover = false;
+  let isCancel = false;
+
+  const onMouseOver = () => {
+    isHover = true;
+  };
+
+  const onMouseLeave = () => {
+    isHover = false;
+    setTimeout(() => {
+      if (!isHover && !isCancel) setActiveHover('');
+      if (isCancel) isCancel = false;
+    }, 2000);
+  };
+
+  const handleClickOutside = () => {
+    isCancel = true;
+    setActiveHover('');
+  };
+
+  const useOutsideClick = (callback: () => void) => {
+    const ref = React.useRef<any>();
+    React.useEffect(() => {
```

Comment:
A `useEffect` in a function is generally a bad practice.  Linting might complain.  I'm not sure it's required.  Just follow the standard practice.

Reply:
Updated, thanks